src/osd/PrimaryLogPG.cc

```diff
@@ -2820,6 +2895,12 @@ void PrimaryLogPG::finish_proxy_read(hobject_t oid, ceph_tid_t tid, int r)
   q->second.erase(it);
   if (q->second.size() == 0) {
     in_progress_proxy_ops.erase(oid);
```

Comment:
shouldn't this go below so it also gets erase in the multiple read completion case?

Reply:
In the multiple read case (client request an single op but it needs several chunks), 
q->second.size() will be > 0 (and just return without read completion) if reading chunks is not finished.  q->second.size() == 0 means that this is the last chunk for requested op.